spaceapi/src/utils.rs

```diff
@@ -27,6 +80,7 @@ impl<T> Optional<T> {
         }
     }
 
+    /// Converts from `Optional<T>` to `Optional<&T>`
     pub fn as_ref<'r>(&'r self) -> Optional<&'r T> {
         match *self {
             Optional::Value(ref x) => Optional::Value(x),
```

Comment:
I implemented it similar to `Option` see https://github.com/rust-lang/rust/blob/d03456183e85fe7bd465bbe7c8f67885a2528444/src/libcore/option.rs#L244